api/v1alpha1/dataprotectionapplication_types.go

```diff
@@ -344,6 +344,8 @@ type PodConfig struct {
 	// labels to add to pods
 	// +optional
 	Labels map[string]string `json:"labels,omitempty"`
+	// (do not use warning) nodeSelector field is for backwards compatibility and
```

Comment:
this is also used for velero deployment/pod configuration. Do we want to not allow use there as well?

Reply:
reading design seems a new field will be added to velero config, but it is still not there, right?

I think we should add both this "deprecation" message and velero change in same PR

Reply:
Repository Maintenance and Velero will be as separate PRs as this one is already too complex imo, so don't want to add even more to it. Repo Maintenance PR https://github.com/openshift/oadp-operator/pull/1688 based on this one, so this one needs to be merged first.

Reply:
Actually I will remove this (do not use warning) because it only applies to the node-agent and we may not want to even remove it. It's explained in the design.

Reply:
I have also updated the design document to make the Velero component optional, noting that we may implement it in the future if needed. This decision is based on the fact that PodConfig.NodeSelector should generally be sufficient. Since the Velero pod itself doesn’t perform actual work but primarily manages operations, and only one pod exists per deployment, simple label matching should be adequate for scheduling.